modules/light-clients/08-wasm/internal/ibcwasm/wasm.go

```diff
@@ -0,0 +1,41 @@
+package ibcwasm
+
+import (
+	"cosmossdk.io/collections"
+	storetypes "cosmossdk.io/core/store"
+)
+
+var (
+	vm WasmEngine
+
+	// state management
+	Schema     collections.Schema
+	CodeHashes collections.KeySet[[]byte]
+
+	// CodeHashesKey is the key under which all code hashes are stored
+	CodeHashesKey = collections.NewPrefix(0)
+)
+
+// SetVM sets the wasm VM for the 08-wasm module.
+func SetVM(wasmVM WasmEngine) {
+	vm = wasmVM
+}
+
+// GetVM returns the wasm VM for the 08-wasm module.
+func GetVM() WasmEngine {
+	return vm
+}
```

Comment:
maybe we should panic if the vm is nil? Shouldn't actually happen by not harm rather than running into a nil pointer.

Reply:
I added the check in the `SetVM` function. Sounds to me like a better place to check it?